src/rendervulkan.cpp

```diff
@@ -2142,21 +2176,20 @@ void vulkan_garbage_collect( void )
 {
 	// If we ever made anything calling get_command_buffer() multi-threaded we'd have to rethink this
 	// Probably just differentiate "busy" and "submitted"
+	uint64_t currentSeqNo;
+	VkResult res = vkGetSemaphoreCounterValue(device, g_scratchTimelineSemaphore, &currentSeqNo);
+	assert( res == VK_SUCCESS );
+
 	for ( uint32_t i = 0; i < k_nScratchCmdBufferCount; i++ )
 	{
-		if ( g_scratchCommandBuffers[ i ].busy == true &&
```

Comment:
I think the comment at the top of this file is stale now and could be removed. With how you're checking the timeline value here this function should be safe.